### ISSUE TITLE
KeeShare writing caused by references

### ISSUE THREAD

**Issue by ndurchx** [keepassxreboot/keepassxc#7282]
## Overview
[TIP]:  # ( DO NOT include screenshots of your actual database! )
[NOTE]: # ( Give a BRIEF summary about your problem )
The change of an password which is referenced at an other entry which is located in an keeshare export should trigger the writing of the keeshare file.

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1.  Create an export-only keeshare folder
2.  Create an entry in this folder with a reference at the password field to an other entry
3.  Save
4. Change the password at the origin entry of the reference
5. Save

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
The .share file should be written with the new password from the reference

## Actual Behavior
[NOTE]: # ( Tell us what actually happens )
The .share file is not written and contains the old password. When the share file is written through another action at the share, the new password is written

## Context
[NOTE]: # ( Give us any additional information you may have. )

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - Version 2.6.6
Revision: 9c108b9

[NOTE]: # ( Pick choices based on your environment )
Operating System: Windows

**Comment by 1enn0** [comment]
I stumbled upon this today, as well! Are there any plans of this getting implemented?

**Comment by pfoo** [comment]
Just hit the same issue
Applies at least to password and username

This feature can be usefull as it allows sharing of an entry without sharing the whole folder

**Comment by droidmonkey** [comment]
Keeshare shares at the entry level, you just happen to configure it at the group/folder level. You can simply move a keeshare'd entry anywhere in your database and it will still technically sync!

**Comment by pfoo** [comment]
According to the user guide, this is true for an import group, not for an export group.

Tested right now, moving an entry outside of the export group -> keeshare is not updating

**Comment by droidmonkey** [comment]
Yes that's a good point, sync and import operate at the entry level.

**Comment by MNarath1** [comment]
I think this is the same issue as described in my PR so i added this issue the PR as it should be fixed the PR
